diff --git a/node_modules/react-native-image-crop-picker/android/src/main/java/com/reactnative/ivpusic/imagepicker/Compression.java b/node_modules/react-native-image-crop-picker/android/src/main/java/com/reactnative/ivpusic/imagepicker/Compression.java
index 5ea266c..651b89d 100644
--- a/node_modules/react-native-image-crop-picker/android/src/main/java/com/reactnative/ivpusic/imagepicker/Compression.java
+++ b/node_modules/react-native-image-crop-picker/android/src/main/java/com/reactnative/ivpusic/imagepicker/Compression.java
@@ -55,7 +55,12 @@ class Compression {
         ExifInterface originalExif = new ExifInterface(originalImagePath);
         String originalOrientation = originalExif.getAttribute(ExifInterface.TAG_ORIENTATION);
 
-        bitmap = Bitmap.createScaledBitmap(bitmap, targetWidth, targetHeight, true);
+        if (originalOrientation.equals(String.valueOf(ExifInterface.ORIENTATION_ROTATE_90))
+            || originalOrientation.equals(String.valueOf(ExifInterface.ORIENTATION_ROTATE_270))) {
+          bitmap = Bitmap.createScaledBitmap(bitmap, targetHeight, targetWidth, true);
+        } else {
+          bitmap = Bitmap.createScaledBitmap(bitmap, targetWidth, targetHeight, true);
+        }
 
         File imageDirectory = context.getExternalFilesDir(Environment.DIRECTORY_PICTURES);
 
